project-1/main.py

```diff
@@ -0,0 +1,88 @@
+from pathlib import Path
+from google.oauth2 import service_account
+from gspread_dataframe import set_with_dataframe
+import gspread
+import pandas as pd
+import logging
+
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.StreamHandler()],
+)
+
+
+class Workbook:
+    #  initialize gspread connections 
+    def __init__(self, workbook_name: str, csv_file: Path, sheet_name: str, email: str) -> None:
+        self.workbook_name = workbook_name
+        self.csv_file = csv_file
+        self.sheet_name = sheet_name
+        self.email = email
+
+        self.df = pd.read_csv(self.csv_file)
+
+        # connection parameters
+        scope = [
+            "https://spreadsheets.google.com/feeds",
+            "https://www.googleapis.com/auth/spreadsheets",
+            "https://www.googleapis.com/auth/drive.file",
+            "https://www.googleapis.com/auth/drive",
+        ]
+
+        try:
+            # google cloud platform API credentials
+            creds = service_account.Credentials.from_service_account_file(
+                "./credentials.json", scopes=scope
+            )
+            # create API client
+            self.client = gspread.authorize(creds)
+
+            logging.info("Connected to Google Sheets successfully.")
+        except Exception as e:
+            logging.error(f"Could not connect to Google Sheets: {e}")
```

Comment:
do you want this to fail and continue and you might want to raise an error here?

Reply:
I didn't think about this since I didn't encounter any errors yet. I'll certainly add the raise error handling. Thank you for the suggestion!